docs/use-keyring-file-vault-components.md

```diff
@@ -0,0 +1,155 @@
+# Use the keyring_file and keyring_vault components
+
+Percona Server supports a keyring that enables internal server components to store sensitive information securely for later retrieval. The server uses a manifest to load the component and the component has its own configuration file. The components are stored in the `plugin` directory.
+
+!!! note
+
+    Only one keyring component can be enabled at a time. Enabling multiple keyring components is not supported.
+
+Percona Server supports the following keyring components:
+
+=== "keyring_vault component"
+
+    ### Use the keyring vault component
+
+    The `keyring_vault` component extends the server capabilities and provides an interface for the database with a [HashiCorp Vault](https://www.hashicorp.com/products/vault/data-protection) server to store key and secure encryption keys.
+
+    !!! admonition "See also"
+
+        [Hashicorp Documentation: Installing Vault]
+        
+        [Hashicorp Documentation: Production Hardening]
+
+    The server uses a manifest to load the component and the component has its own configuration file.
+
+    An example of a manifest and a configuration file is the following:
+
+    An example of `./bin/mysqld.my`:
+
+    ```json
+    {
+       "components": "file://component_keyring_vault"
+    }
+    ```
+
+    An example of `/lib/plugin/component_keyring_vault.cnf`:
+
+    ```json
+    {
+       "path": "/var/lib/c/keyring_vault", "read_only": false
+    }
+    ```
+
+    Set the `keyring_vault_config` variable to point to the `keyring_vault` configuration file.
+
+    The `keyring_vault_config` variable defines the location of the `keyring_vault` configuration file.
+
+    `keyring_vault_config`
+
+    | Option       | Description            |
+    |--------------|------------------------|
+    | Command-line | --keyring-vault-config |
+    | Scope        | Global                 |
+    | Dynamic      | Yes                    |
+    | Data type    | Text                   |
+    | Default      |                        |
+
+    The `keyring_vault_config` file contains the following information:
```

Comment:
There is no `keyring_vault_config` variable any more. Component config search is done similar to manifest file.
1) The component_keyring_vault config name is expected to be "component_keyring_vault.cnf".
2) There may be two configs - global one (located in 'plugin' directory), and instance specific (located in data directory).
3) Global config may have "read_local_config" option set which indicates server should search for instance specific config.

Reply:
Rewritten